### DISCUSSION TITLE
Feedback collection

### DISCUSSION THREAD

**Discussion by Daydreamer-riri** [Daydreamer-riri/vscode-ext-packages#1] [General]
Hi, @childrentime , I have developed an interesting vscode extension. Due to my technical background, I have rarely used package management tools other than `npm` and `pnpm`, so I lack knowledge about the protocols of other package management tools. I hope you can try this extension and see if there are any issues.

**Comment by childrentime** [comment]
Well. First I think there  haven't any connection between reading package.json and package management tools. I found two issues: The first issue is that scrolling the package `json` file gets stuck when installing plugins if there are too many dependencies listed in the file. The second issue happens when the version of the dependency is '*', instead of specifying a specific version, which causes it to return `undefined`.

**Reply by Daydreamer-riri** [reply]
Regarding your first point, it does indeed take quite a long time (possibly several minutes) to obtain versions of a large number of packages, which is the issue I hope to address. As for the second point, I have not yet come across any issues related to '*'.

Then regarding the package manager, what I actually mean is a protocol similar to `workspace:*` in pnpm.

**Reply by childrentime** [reply]
@Daydreamer-riri just like "react: *"， which means we can accpet any versions of react, will return undefined

**Reply by Daydreamer-riri** [reply]
Currently, the package version is processed by semver, and "*" will be matched by semver as the latest version

**Comment by childrentime** [comment]
`const { spawn } = require('child_process');`

The child_process.exec() method creates a new process each time it's called to execute a command. Every time it's called, a new child process is created and the specified command is executed within it. When the command is done executing, the child process exits. If you need to execute multiple commands in your Node.js application and retain interactive capabilities with the subprocesses, consider using the child_process.spawn() method. The spawn() method starts a new process and allows you to interact with its standard input/output in real-time in your program.

**Reply by Daydreamer-riri** [reply]
Will using the same child process prevent parallel fetching of multiple package versions? However, creating a large number of child processes each time may indeed lead to performance issues.

**Reply by childrentime** [reply]
Use '&'? I haven't tried it either.
